pygluu/kubernetes/gui/views/wizard.py

```diff
@@ -1303,22 +1199,27 @@ def upgrade():
         # get supported versions image name and tag
         versions, version_number = get_supported_versions()
         image_names_and_tags = versions.get(gluu_settings.db.get("GLUU_UPGRADE_TARGET_VERSION"), {})
+        custom_images = []
         for k, v in image_names_and_tags.items():
             if "IMAGE_NAME" in k and gluu_settings.db.get(k) != v:
-                continue
-            gluu_settings.db.set(k, v)
-            
-        return redirect(url_for(request.form["next_step"]))
+                image = '_'.join(k.split("_")[:2])
+                custom_images.append(f"{image}_NAME")
+                custom_images.append(f"{image}_TAG")
+
+        for k, v in image_names_and_tags.items():
+            if k not in custom_images:
```

Comment:
When users are using custom image tag, the `custom_images` are empty hence the tag will be overwritten. Fortunately, users can re-enter their custom images name and/or tag.

Reply:
From my point of view, the `/gluu-upgrade` wizard should have warning about image names/tags being overwritten, also notify users that they will be able to define custom image names/tags in next step.